ProcessLib/HydroMechanics/HydroMechanicsFEM-impl.h

```diff
@@ -221,15 +225,25 @@ void HydroMechanicsLocalAssembler<ShapeFunctionDisplacement,
         auto& eps = _ip_data[ip].eps;
         auto const& sigma_eff = _ip_data[ip].sigma_eff;
 
-        vars[static_cast<int>(MPL::Variable::phase_pressure)] = N_p.dot(p);
+        double const p_int_pt = N_p.dot(p);
+        vars[static_cast<int>(MPL::Variable::phase_pressure)] = p_int_pt;
 
         auto const K_S = solid_material.getBulkModulus(t, x_position);
 
+        auto const alpha = solid.property(MPL::PropertyType::biot_coefficient)
+                               .template value<double>(vars, x_position, t, dt);
+
+        // For stress dependent permeability.
+        // TODO (wenqing): get a model type, and use it to avoid the following
+        // assignment if no stress dependent permeability model is used.
```

Comment:
Just to get your idea, how would you do it? How would you find out if an arbitrary permeability model is stress dependent?

Reply:
Can we add a member to Property to return its type?

Reply:
I don't quite get it. Would you elaborate?
So there will be a member in the Property class, let's say x(). It would return what?
How would you use the result here to find out if an *arbitrary* permeability model is stress dependent?

Reply:
For example:
```
virtual bool Property::hasStressVariable() const {return false;}

 bool PermeabilityMohrCoulombFailureIndexModel::hasStressVariable() const override {return true;}
 ```

Reply:
This does not scale well, because for every new variable like stress etc. the Property interface has to be updated. So, then one would add a new enum value, two string conversions (back and forth), and additionally a test function and an if-condition to every process which is going to use it. Also how do you find out for MFront materials if there is dependency on a particular variable?
I'd drop the comment. Keeping things simple is an advantage. How much time would it save for each assembly?

Reply:
The major computation is a two vector (with a size of 4 or 6) addition at the integration points. I have not compared the time increases in the local assembler for large problems. I think, it may increase the local assembly time for the maximum magnitude of 1%. 

An enum type might be good. This would be put to another PR if an if-condition for passing stress to Property is needed.